File: aldrin/src/main/java/ovh/testdomain/project/models/Visit.java
package ovh.testdomain.project.models;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.validation.constraints.NotBlank;
import java.util.Date;

@Entity
public class Visit {
    @Id
    @GeneratedValue
    private Long id;

    @NotBlank
    private String ip;

    @NotBlank
    private Date timestamp;


    public Visit() {
        setTimestamp(new Date());
    }

    public void setIp(String ip) {
        this.ip = ip;
    }

    public String getIp() {
        return ip;
    }

    private void setTimestamp(Date timestamp) {
        this.timestamp = timestamp;
    }

    public Date getTimestamp() {
        return timestamp;
    }
}

File: titov/src/Entity/Visit.php
<?php

namespace App\Entity;

use DateTime;
use DateTimeInterface;
use Symfony\Component\Validator\Constraints as Assert;
use App\Elastic\Schema;

/**
 * @Schema\Index(name="visit", alias="visit_alias")
 */
class Visit extends Entity
{
    /**
     * @var string
     * @Schema\Field(type="string", index="not_analyzed")
     * @Assert\NotBlank
     */
    protected $ip;

    /**
     * @var DateTimeInterface
     * @Schema\Field(type="date", format="dateOptionalTime")
     * @Assert\NotBlank
     */
    protected $timestamp;

    public function __construct()
    {
        $this->setTimestamp(new DateTime());
    }

    public function getIp(): ?string
    {
        return $this->ip;
    }

    public function setIp(?string $value): self
    {
        $this->ip = $value;
        return $this;
    }

    public function getTimestamp(): DateTimeInterface
    {
        return $this->timestamp;
    }

    private function setTimestamp(DateTimeInterface $value): self
    {
        $this->timestamp = $value;
        return $this;
    }
}
